lib/phoenix_html/link.ex

```diff
@@ -165,4 +175,11 @@ defmodule Phoenix.HTML.Link do
 
     {value, opts}
   end
+
+  defp invalid_destination?(to) when is_tuple(to), do: false
```

Comment:
This check would allow this to pass: `link "Foo", to: html_escape(conn.params["bar])`. Is this the intent? If so, we need a test for it and we should probably tweak the message above by asking developers to use `html_escape(javascript)` instead of `{:javascript, rest}`.

Reply:
I believe the "to" value would already be escaped, so there wouldn't really be a need for them to `html_escape` that input. However, I do think this should *technically* work. Depending on your thoughts for transforming the "to" value vs raising, this tuple check will end up doing different things. But, if you like the transformation, I think the check would allow arbitrary schemes. So it could look something like this..

```
defp destination({scheme, to}) when is_atom(scheme) do
  if valid_destination?("#{scheme}:", to), do: to
end

defp valid_destination?(scheme, scheme <> _), do: true 
```

Consequently, `link "Foo", to: html_escape(conn.params["bar])` would work, but it would result in output like `<a href="safe:user_input">Foo</a>`.